src/main/kotlin/org/rust/ide/inspections/RsSelfConventionInspection.kt

```diff
@@ -19,12 +19,13 @@ class RsSelfConventionInspection : RsLocalInspectionTool() {
     override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) =
         object : RsVisitor() {
             override fun visitFunction(m: RsFunction) {
-                // Should this handle traits as well perhaps?
-                if (m.owner !is RsFunctionOwner.Impl) return
+                val owner = m.owner
+                if (!owner.isImplOrTrait) return
+                if (owner.isTraitImpl) return
```

Comment:
I think it would be more readable if all owner checks were consolidated in a single if, like `if (!(owner.isInherentImpl || owner.isTrait)) return`